Issuetracker : #i53288#
CWS          : N/A
Author       : <maho@openoffice.org> (JCA)
Description  : remove circular dependency

in this case,
core01 is depend on core02-core08, and core0[2-9] and core10
depends on core01. for FreeBSD this is regarded as circular dependency
and pkg_add command is forked infinite times and eats up systems' resource.
namely when core01 tries to install core02, core02 also tries to
install core01.
Index: instsetoo_native/inc_openoffice/unix/packagelist.txt
===================================================================
RCS file: /cvs/installation/instsetoo_native/inc_openoffice/unix/packagelist.txt,v
retrieving revision 1.21
diff -u -r1.21 packagelist.txt
--- instsetoo_native/inc_openoffice/unix/packagelist.txt	30 Nov 2005 13:21:44 -0000	1.21
+++ instsetoo_native/inc_openoffice/unix/packagelist.txt	3 Dec 2005 22:26:12 -0000
@@ -37,6 +37,7 @@
 packagename = "%UNIXPRODUCTNAME-core01"
 linuxreplaces = "%WITHOUTDOTUNIXPRODUCTNAME-mailcap,%WITHOUTDOTUNIXPRODUCTNAME-core,%WITHOUTDOTUNIXPRODUCTNAME-core01"
 requires = "%UNIXPRODUCTNAME-core02,%UNIXPRODUCTNAME-core03,%UNIXPRODUCTNAME-core04,%UNIXPRODUCTNAME-core05,%UNIXPRODUCTNAME-core06,%UNIXPRODUCTNAME-core07,%UNIXPRODUCTNAME-core08"
+freebsdrequires = "%UNIXPRODUCTNAME-core02,%UNIXPRODUCTNAME-core03,%UNIXPRODUCTNAME-core04,%UNIXPRODUCTNAME-core05,%UNIXPRODUCTNAME-core06,%UNIXPRODUCTNAME-core07,%UNIXPRODUCTNAME-core08,%UNIXPRODUCTNAME-core09,%UNIXPRODUCTNAME-core10"
 copyright = "1999-2005 by OpenOffice.org"
 vendor = "OpenOffice.org"
 description = "Core module for %PRODUCTNAME %PRODUCTVERSION"
@@ -278,6 +279,7 @@
 packagename = "%UNIXPRODUCTNAME-core02"
 linuxreplaces = "%WITHOUTDOTUNIXPRODUCTNAME-core02"
 requires = "%UNIXPRODUCTNAME-core01"
+freebsdrequires = ""
 copyright = "1999-2005 by OpenOffice.org"
 vendor = "OpenOffice.org"
 description = "Office core module for %PRODUCTNAME %PRODUCTVERSION"
@@ -291,6 +293,7 @@
 packagename = "%UNIXPRODUCTNAME-core03"
 linuxreplaces = "%WITHOUTDOTUNIXPRODUCTNAME-core03"
 requires = "%UNIXPRODUCTNAME-core01"
+freebsdrequires = ""
 copyright = "1999-2005 by OpenOffice.org"
 vendor = "OpenOffice.org"
 description = "Office core module for %PRODUCTNAME %PRODUCTVERSION"
@@ -304,6 +307,7 @@
 packagename = "%UNIXPRODUCTNAME-core04"
 linuxreplaces = "%WITHOUTDOTUNIXPRODUCTNAME-core04"
 requires = "%UNIXPRODUCTNAME-core01"
+freebsdrequires = ""
 copyright = "1999-2005 by OpenOffice.org"
 vendor = "OpenOffice.org"
 description = "Office core module for %PRODUCTNAME %PRODUCTVERSION"
@@ -317,6 +321,7 @@
 packagename = "%UNIXPRODUCTNAME-core05"
 linuxreplaces = "%WITHOUTDOTUNIXPRODUCTNAME-core05"
 requires = "%UNIXPRODUCTNAME-core01"
+freebsdrequires = ""
 copyright = "1999-2005 by OpenOffice.org"
 vendor = "OpenOffice.org"
 description = "Office core module for %PRODUCTNAME %PRODUCTVERSION"
@@ -330,6 +335,7 @@
 packagename = "%UNIXPRODUCTNAME-core06"
 linuxreplaces = "%WITHOUTDOTUNIXPRODUCTNAME-core06"
 requires = "%UNIXPRODUCTNAME-core01"
+freebsdrequires = ""
 copyright = "1999-2005 by OpenOffice.org"
 vendor = "OpenOffice.org"
 description = "Office core module for %PRODUCTNAME %PRODUCTVERSION"
@@ -343,6 +349,7 @@
 packagename = "%UNIXPRODUCTNAME-core07"
 linuxreplaces = "%WITHOUTDOTUNIXPRODUCTNAME-core07"
 requires = "%UNIXPRODUCTNAME-core01"
+freebsdrequires = ""
 copyright = "1999-2005 by OpenOffice.org"
 vendor = "OpenOffice.org"
 description = "Office core module for %PRODUCTNAME %PRODUCTVERSION"
@@ -356,6 +363,7 @@
 packagename = "%UNIXPRODUCTNAME-core08"
 linuxreplaces = "%WITHOUTDOTUNIXPRODUCTNAME-core08"
 requires = "%UNIXPRODUCTNAME-core01"
+freebsdrequires = ""
 copyright = "1999-2005 by OpenOffice.org"
 vendor = "OpenOffice.org"
 description = "Office core module for %PRODUCTNAME %PRODUCTVERSION"
@@ -369,6 +377,7 @@
 packagename = "%UNIXPRODUCTNAME-core09"
 linuxreplaces = "%WITHOUTDOTUNIXPRODUCTNAME-core09"
 requires = "%UNIXPRODUCTNAME-core01"
+freebsdrequires = ""
 copyright = "1999-2005 by OpenOffice.org"
 vendor = "OpenOffice.org"
 description = "Office core module for %PRODUCTNAME %PRODUCTVERSION"
@@ -383,6 +392,7 @@
 packagename = "%UNIXPRODUCTNAME-core10"
 linuxreplaces = "%WITHOUTDOTUNIXPRODUCTNAME-core10"
 requires = "%UNIXPRODUCTNAME-core01"
+freebsdrequires = ""
 copyright = "1999-2005 by OpenOffice.org"
 vendor = "OpenOffice.org"
 description = "Office core module for %PRODUCTNAME %PRODUCTVERSION"
Index: solenv/bin/modules/installer/epmfile.pm
===================================================================
RCS file: /cvs/tools/solenv/bin/modules/installer/epmfile.pm,v
retrieving revision 1.42
diff -u -r1.42 epmfile.pm
--- solenv/bin/modules/installer/epmfile.pm	9 Nov 2005 09:09:49 -0000	1.42
+++ solenv/bin/modules/installer/epmfile.pm	3 Dec 2005 22:26:52 -0000
@@ -395,6 +395,11 @@
 		$provides = "solarisprovides";	 # the name in the packagelist	
 		$requires = "solarisrequires";	 # the name in the packagelist	
 	}
+	elsif ( $installer::globals::isfreebsdpkgbuild )
+	{
+		$provides = "freebsdprovides";	 # the name in the packagelist	
+		$requires = "freebsdrequires";	 # the name in the packagelist	
+	}
 	else
 	{
 		$provides = "provides";	 		# the name in the packagelist	
Index: solenv/bin/modules/installer/globals.pm
===================================================================
RCS file: /cvs/tools/solenv/bin/modules/installer/globals.pm,v
retrieving revision 1.47
diff -u -r1.47 globals.pm
--- solenv/bin/modules/installer/globals.pm	9 Nov 2005 09:10:01 -0000	1.47
+++ solenv/bin/modules/installer/globals.pm	3 Dec 2005 22:26:52 -0000
@@ -80,6 +80,7 @@
 	$issolarispkgbuild = 0;
 	$issolarissparcbuild = 0;
 	$issolarisx86build = 0;
+	$isfreebsdpkgbuild = 0;
 	$unpackpath = "";
 	$idttemplatepath = "";
 	$idtlanguagepath = "";

Index: solenv/bin/modules/installer/parameter.pm
===================================================================
RCS file: /cvs/tools/solenv/bin/modules/installer/parameter.pm,v
retrieving revision 1.29
diff -u -r1.29 parameter.pm
--- solenv/bin/modules/installer/parameter.pm	9 Nov 2005 09:10:12 -0000	1.29
+++ solenv/bin/modules/installer/parameter.pm	10 Dec 2005 13:11:30 -0000
@@ -304,6 +304,13 @@
 			$installer::globals::epmoutpath = "packages";
 		}
 	}
+	if ( $installer::globals::compiler =~ /unxfbsdi/ )
+	{
+		if ( $installer::globals::packageformat eq "bsd" )
+		{
+			$installer::globals::isfreebsdpkgbuild = 1;
+		}
+	}
 
 	if ( $installer::globals::compiler =~ /unxso[lg]s/ ) { $installer::globals::issolarissparcbuild = 1; }
 	
